tests/conftest.py

```diff
@@ -55,11 +78,93 @@ def db_fixture_dir(pytestconfig) -> Path:
     return pytestconfig.rootpath.joinpath("db", "fixtures")
 
 
+@pytest.fixture(scope="session")
+def io_fixture_dir(pytestconfig) -> Path:
+    return pytestconfig.rootpath.joinpath("io", "fixtures")
+
+
 @pytest.fixture(scope="session")
 def settings() -> Settings:
     return Settings()
 
 
+def generate_io_fixture(fixtures, io_postfix=None, scope="module"):
+    @pytest.fixture(scope=scope)
+    def io_fixture(io_fixture_dir, settings):
+        nonlocal io_postfix
+        nonlocal fixtures
+
+        io_postfix = (
+            "".join(random.choices(string.ascii_letters, k=5))
+            if not io_postfix
+            else io_postfix
+        )
+        bucket_name = (
+            str(settings.bucket_name + "-" + io_postfix)
+            .replace(".", "-")
+            .replace("_", "-")
+        )
+
+        def setup_io():
+            ioclient = IOClient(
+                settings.s3_endpoint,
+                settings.access_key_id,
+                settings.secret_access_key,
+                bucket_name,
+            )
+            for fixture in fixtures:
+                path = io_fixture_dir.joinpath(fixture["source"])
+                if not path.is_dir():
+                    raise ValueError(f"Unknown fixture '{str(path)}'")
+                files = Path(path).glob("*")
+                for file in files:
+                    if not file.is_file():
+                        raise ValueError(f"Unknown fixture file '{str(file)}'")
+                    load_jsonfile(
+                        ioclient,
+                        "{}/{}".format(fixture["destination"], file.name),
+                        str(file),
+                    )
+
+        def cleanup_io():
+            ioclient = IOClient(
+                settings.s3_endpoint,
+                settings.access_key_id,
+                settings.secret_access_key,
+                bucket_name,
+            )
+            for fixture in fixtures:
+                path = io_fixture_dir.joinpath(fixture["source"])
+                if not path.is_dir():
+                    raise ValueError(f"Unknown fixture '{str(path)}'")
+                files = Path(path).glob("*")
+                for file in files:
+                    if file.is_file():
+                        remove_object(
+                            ioclient, "{}/{}".format(fixture["destination"], file.name)
+                        )
+
+        try:
+            create_bucket(
+                settings.s3_endpoint,
+                settings.access_key_id,
+                settings.secret_access_key,
+                bucket_name,
+            )
+            setup_io()
+            yield bucket_name
+        finally:
+            cleanup_io()
```

Comment:
As we discussed, it would likely make sense to make this a method on the `IOClient` class.

Reply:
Fixed.